crates/pindakaas/src/helpers.rs

```diff
@@ -221,30 +282,20 @@ pub(crate) mod tests {
 
 	use crate::{
 		bool_linear::BoolLinExp,
-		integer::IntVarEnc,
+		integer::IntVar,
 		solver::{cadical::Cadical, SolveResult, Solver},
-		Checker, ClauseDatabaseTools, Cnf, Lit, Valuation,
+		Checker, ClauseDatabase, ClauseDatabaseTools, Cnf, Lit, Valuation, Var, VarRange,
 	};
 
 	/// Helper functions to ensure that the possible solutions of a formula
 	/// abide by the given checker.
 	pub(crate) fn assert_checker(formula: &Cnf, checker: &impl Checker) {
-		let mut slv = Cadical::from(formula);
-		let vars = formula.get_variables();
-		while let SolveResult::Satisfied(value) = slv.solve() {
-			assert_eq!(checker.check(&value), Ok(()));
-			let no_good: Vec<Lit> = vars
-				.map(|v| {
-					let l = v.into();
-					if value.value(l) {
-						!l
-					} else {
-						l
-					}
-				})
-				.collect();
-			slv.add_clause(no_good).unwrap();
-		}
+		Cadical::from(formula)
+			.solve_all(formula.get_variables())
```

Comment:
I've implemented a solve_all which is nice to have, however, it's not yet as iter so it finds all solutions first (this is a regression over the old way of course). I still suggest making this an issue rather than fixing it in this PR.